sickbeard/subtitles.py

```diff
@@ -78,266 +96,546 @@ def sorted_service_list():
+def _get_video(video_path, subtitles_dir, subtitles, embedded_subtitles, release_name):
+    """Internal get_video method since dogpile cache default function_key_generator doesn't accept keyword arguments
+    """
     try:
+        video_path = encode(video_path)
+        subtitles_dir = encode(subtitles_dir or get_subtitles_dir(video_path))
+
+        logger.log(u'Scanning video {0}...'.format(video_path), logger.DEBUG)
         video = scan_video(video_path)
 
         # external subtitles
         if subtitles:
-            video.subtitle_languages |= set(search_external_subtitles(video_path, directory=subtitles_path).values())
+            video.subtitle_languages |= set(search_external_subtitles(video_path, directory=subtitles_dir).values())
 
         if embedded_subtitles is None:
             embedded_subtitles = bool(not sickbeard.EMBEDDED_SUBTITLES_ALL and video_path.endswith('.mkv'))
 
         refine(video, episode_refiners=episode_refiners, embedded_subtitles=embedded_subtitles,
                release_name=release_name)
+        return video
     except Exception as error:
-        logger.log(u'Exception: {}'.format(error), logger.DEBUG)
-        return None
+        logger.log(u'Exception: {0}'.format(error), logger.DEBUG)
 
-    return video
 
+def get_subtitles_dir(video_path):
+    """Returns the correct subtitles directory based on the user configuration.
+
+    If the directory doesn't exist, it will be created
+
+    :param video_path: the video path
+    :type video_path: str
+    :return: the subtitles directory
+    :rtype: str
+    """
+    if not sickbeard.SUBTITLES_DIR:
+        return os.path.dirname(video_path)
 
-def get_subtitles_path(video_path):
     if os.path.isabs(sickbeard.SUBTITLES_DIR):
-        new_subtitles_path = sickbeard.SUBTITLES_DIR
-    elif sickbeard.SUBTITLES_DIR:
-        new_subtitles_path = os.path.join(os.path.dirname(video_path), sickbeard.SUBTITLES_DIR)
-        dir_exists = sickbeard.helpers.makeDir(new_subtitles_path)
-        if not dir_exists:
-            logger.log(u'Unable to create subtitles folder {}'.format(new_subtitles_path), logger.ERROR)
-        else:
-            sickbeard.helpers.chmodAsParent(new_subtitles_path)
-    else:
-        new_subtitles_path = os.path.dirname(video_path)
+        return sickbeard.SUBTITLES_DIR
 
-    try:
-        # Encode path to UTF-8 to ensure subliminal support.
-        new_subtitles_path = new_subtitles_path.encode('utf-8')
-    except UnicodeEncodeError:
-        # Fallback to system encoding. This should never happen.
-        new_subtitles_path = new_subtitles_path.encode(sickbeard.SYS_ENCODING)
+    new_subtitles_path = os.path.join(os.path.dirname(video_path), sickbeard.SUBTITLES_DIR)
+    if sickbeard.helpers.makeDir(new_subtitles_path):
+        sickbeard.helpers.chmodAsParent(new_subtitles_path)
+    else:
+        logger.log(u'Unable to create subtitles folder {0}'.format(new_subtitles_path), logger.ERROR)
 
```

Comment:
Sure... I'll change it later today.
The [old code](https://github.com/pymedusa/SickRage/pull/219/files?diff=split#diff-425be5582fb3c7b0fce3dbcd319fed75L309) was ERROR as well: